include/drivers/gpio/gpio_sx1509b.h

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_GPIO_GPIO_SX1509B_H_
+#define ZEPHYR_INCLUDE_DRIVERS_GPIO_GPIO_SX1509B_H_
+
+int sx1509b_pwm_pin_configure(struct device *dev, gpio_pin_t pin);
+
+int sx1509b_pwm_pin_set(struct device *dev, gpio_pin_t pin, uint8_t pwm_value);
```

Comment:
More documentation here.  What does `pwm_value` mean (there is no such register in the data sheet)?  How does one figure out what value to pass?

Oh, I see: this maps to the IOn register which is intensity.  Maybe use that name instead of "pwm" which in Zephyr is usually defined as a period and a pulse width.